src/board.rs

```diff
@@ -87,40 +87,104 @@ impl Board {
         Ok(())
     }
 
-    fn add_moves_in_direction(&mut self, start: Position, m: Move) {
-        let mut position = start + m;
-        while Self::is_in_bounds(position).is_ok() {
-            if let Some(piece) = self.get_piece(position) {
-                // allow capturing an opponent's piece
-                if piece.get_player() != self.player {
-                    self.moves.insert((start, position));
+    fn add_pawn_advance_moves(&mut self, start: Position, player: Player) {
+        let direction = match player {
+            Player::White => -1,
+            Player::Black => 1,
+        };
+        let advance_moves = match player {
+            Player::White => Move::get_pawn_advance_moves_white(),
+            Player::Black => Move::get_pawn_advance_moves_black(),
+        };
+
+        for &m in advance_moves {
+            let new_position = Position {
+                x: start.x + (direction * m.dx) as usize,
+                y: start.y + m.dy as usize,
+            };
+            if Self::is_in_bounds(new_position).is_ok() && self.get_piece(new_position).is_none() {
+                self.moves.insert((start, new_position));
+                if ((player == Player::White && start.x == 6)
+                    || (player == Player::Black && start.x == 1))
+                    && self
+                        .get_piece(Position {
```

Comment:
Maybe we could define the right multiply operators and this could be `start + 2 * m`.

Reply:
ok